crates/ruff_linter/resources/test/fixtures/flake8_pyi/PYI016.py

```diff
@@ -111,3 +111,8 @@ def func2() -> str | str:  # PYI016: Duplicate union member `str`
 
 # Test case for mixed union type
 field34: typing.Union[list[int], str] | typing.Union[bytes, list[int]]  # Error
+
+
+field35: "int | str" | int  # Error, but currently not detected
```

Comment:
I'm not sure I'd want to extend the rule to detect this, since:
- It will fail at runtime with `TypeError`
- In a stub file it won't fail, but you shouldn't ever use quoted annotations in a stub file anyway, and we have PYI020 to tell you off about that if you do

So I don't think it would be worth the added complexity